app/models/Consignor.scala

```diff
@@ -36,7 +36,22 @@ object Consignor {
 
   implicit lazy val format: OFormat[Consignor] = Json.format[Consignor]
 
-  implicit val xmlReader: XmlReader[Consignor] = {
+  implicit lazy val xmlReader: XmlReader[Consignor] =
+    xmlReaderGoodsLevel.or(xmlReaderHeaderLevel)
+
+  private val xmlReaderGoodsLevel: XmlReader[Consignor] = {
+    (
+      (__ \ "NamCO27").read[String],
+      (__ \ "StrAndNumCO222").read[String],
+      (__ \ "PosCodCO223").read[String],
+      (__ \ "CitCO224").read[String],
+      (__ \ "CouCO225").read[String],
+      (__ \ "NADLNGGTCO").read[String].optional,
+      (__ \ "TINCO259").read[String].optional
+    ).mapN(apply)
+  }
+
+  private val xmlReaderHeaderLevel: XmlReader[Consignor] = {
```

Comment:
Might be safer here to manually call the reader rather than rely on the implicit reader to figure out if its head level or goods level. Though this will also work fine.

Reply:
Nice one, I've updated it to call directly which reader it wants. Have a look and let me know what you think. I've also pushed another change to limit street for Consignee (matches behaviour of Consignor and it's also what the xsd states)